sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs

```diff
@@ -300,5 +300,55 @@ private static void AddTelemetryFromActivityEvents(Activity activity, TelemetryI
                 BaseData = exceptionData,
             };
         }
+
+        internal static void AddEnqueuedTimeToMeasurements(Activity activity, IDictionary<string, double> measurements)
+        {
+            if (activity.Links != null && activity.Links.Any())
+            {
+                if (TryGetAverageTimeInQueueForBatch(activity.Links, activity.StartTimeUtc, out long enqueuedTime))
+                {
+                    measurements["timeSinceEnqueued"] = enqueuedTime;
+                }
+            }
+        }
+
+        private static bool TryGetAverageTimeInQueueForBatch(IEnumerable<ActivityLink> links, DateTimeOffset requestStartTime, out long avgTimeInQueue)
+        {
+            avgTimeInQueue = 0;
+            var linksCount = 0;
+            long startEpochTime = requestStartTime.ToUnixTimeMilliseconds();
+            foreach (ActivityLink link in links)
```

Comment:
Pass activity and use `EnumerateLinks`
```suggestion
            foreach (ref readonly var link in activity.EnumerateLinks())
```